### PR TITLE
Adds mechanism to prevent spam from unhealthy notifications.

### PR DESCRIPTION
**Pull request by GustavoFonsecaDeAlmeida**
**What this PR does / why we need it**:
Adds mechanism to prevent spam from unhealthy notifications.
**Which issue(s) this PR fixes**:
None

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [ ] Created/updated tests
- [x] Unit tests passing
- [x] End-to-end tests passing
- [x] Extended the documentation
- [x] Provided sample for the feature

### CODE REVIEW

File: src/HealthChecks.UI/Configuration/Settings.cs [hunk 1]
```diff
@@ -50,6 +51,11 @@ public Settings DisableDatabaseMigrations()
             DisableMigrations = true;
             return this;
         }
+
+        public Settings NotifyUnHealthyOnceTimeUntilChangeToAvoideNotificationSpam() {
```
Comment (sungam3r) [inline src/HealthChecks.UI/Configuration/Settings.cs:55]:
```suggestion
        public Settings NotifyUnHealthyOnceTimeUntilChangeToAvoidNotificationSpam() {
```

File: src/HealthChecks.UI/Core/HostedService/HealthCheckReportCollector.cs [hunk 2]
```diff
@@ -148,6 +151,32 @@ private async Task<HealthCheckExecution> GetHealthCheckExecution(HealthCheckConf
                 .Where(le => le.Name == configuration.Name)
                 .SingleOrDefaultAsync();
         }
+
+        private async Task<bool> ShouldNotify(string healthCheckName , UIHealthReport report )
```
Comment (sungam3r) [inline src/HealthChecks.UI/Core/HostedService/HealthCheckReportCollector.cs:155]:
```suggestion
        private async Task<bool> ShouldNotify(string healthCheckName , UIHealthReport report)

```

File: src/HealthChecks.UI/Core/HostedService/HealthCheckReportCollector.cs [hunk 3]
```diff
@@ -148,6 +151,32 @@ private async Task<HealthCheckExecution> GetHealthCheckExecution(HealthCheckConf
...
+
+            if (lastNotifications.Count == 2)
+            {
+                var first = lastNotifications[0];
+                var second = lastNotifications[1];
+                if (first.IsUpAndRunning == second.IsUpAndRunning)
+                {
+                    return false;
+                }
+
```
Comment (sungam3r) [inline src/HealthChecks.UI/Core/HostedService/HealthCheckReportCollector.cs:175]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -459,6 +459,20 @@ Webhooks can be configured with configuration providers and also by code. Using
...
+**Avoid Fail notification spam**
+
+To prevent you from receiving several failure notifications from your application, a configuration was created to meet this scenario.
+
+```csharp
+
+services.AddHealthChecksUI(setup =>
+{
+    setup.NotifyUnHealthyOnceTimeUntilChangeToAvoideNotificationSpam(); //You will only receive one failure notification until the status changes.
+
```
Comment (sungam3r) [inline README.md:471]:
```suggestion
    setup.NotifyUnHealthyOnceTimeUntilChangeToAvoidNotificationSpam(); //You will only receive one failure notification until the status changes.
```

**Comment by unaizorrilla** [comment]
Hi 

Thanks for contribute this, @CarlosLanderas  is on vacation but this will be reviewed soon!

unai

**Comment by GustavoFonsecaDeAlmeida** [comment]
GN!! Any news about ?

**Comment by unaizorrilla** [comment]
Hi,

Thanks for contribute this 

We are on vacation right now, let some days to check and review

**Comment by skrty** [comment]
Hi, any news here?

**Comment by sungam3r** [comment]
@GustavoFonsecaDeAlmeida Please rebase onto `master` if you are going to continue work on this PR. Then I can review/merge.

**Comment by sungam3r** [comment]
ping @GustavoFonsecaDeAlmeida

**Comment by sungam3r** [comment]
@GustavoFonsecaDeAlmeida Do you intend to continue working on that PR?

**Comment by sungam3r** [comment]
Thanks @GustavoFonsecaDeAlmeida , superseded by #1877

**Pull request closed without merging**